solenspipe/solenspipe.py

```diff
@@ -20,6 +20,105 @@
 config = io.config_from_yaml(os.path.dirname(os.path.abspath(__file__)) + "/../input/config.yml")
 opath = config['data_path']
 
+def four_split_phi(Xdat_0,Xdat_1,Xdat_2,Xdat_3,Xdatp_0=None,Xdatp_1=None,Xdatp_2=None,Xdatp_3=None,q_func1=None,qfunc2=None):
+    """Return kappa_alms combinations required for the 4cross estimator.
+
+    Args:
+        Xdat_0 (array): [fTalm,fEalm,fBalm] list of filtered alms from split 0
+        Xdat_1 (array): [fTalm,fEalm,fBalm] list of filtered alms from split 1
+        Xdat_2 (array): [fTalm,fEalm,fBalm] list of filtered alms from split 2
+        Xdat_3 (array): [fTalm,fEalm,fBalm] list of filtered alms from split 3
+        q_func1 (function): function for quadratic estimator
+        Xdatp_0 (array): [fTalm,fEalm,fBalm] list of filtered alms from split 0 used for RDN0 for different sim data combination
+        Xdatp_1 (array): [fTalm,fEalm,fBalm] list of filtered alms from split 1 used for RDN0 for different sim data combination
+        Xdatp_2 (array): [fTalm,fEalm,fBalm] list of filtered alms from split 2 used for RDN0 for different sim data combination
+        Xdatp_3 (array): [fTalm,fEalm,fBalm] list of filtered alms from split 3 used for RDN0 for different sim data combination
+        qfunc2 ([type], optional): [description]. Defaults to None.
+
+    Returns:
+        array: Combination of reconstructed kappa alms
+    """
+    q_bh_1=q_func1
+    if Xdatp_0 is None:
```

Comment:
Would be useful to add a reference here to the equations this code is implementing

Reply:
done in latest commit